src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/edit-polling-source.service.ts

```diff
@@ -0,0 +1,176 @@
+import { parse } from "yaml";
+import {
+    EditFormParseType,
+    EditFormType,
+    FetchKind,
+    MergeKind,
+    NameValue,
+} from "./add-polling-source-form.types";
+import { SetPollingSourceSection } from "src/app/shared/shared.types";
+import { FormArray, FormBuilder, FormGroup } from "@angular/forms";
+import { RxwebValidators } from "@rxweb/reactive-form-validators";
+
+@Injectable({
+    providedIn: "root",
+})
+export class EditPollingSourceService {
+    private currentPage = 0;
+    private historyLength = 200;
+    public history: DatasetHistoryUpdate;
+
+    constructor(
+        private appDatasetService: DatasetService,
+        private blockService: BlockService,
+        private appDatasetSubsService: AppDatasetSubscriptionsService,
+        private fb: FormBuilder,
+    ) {}
+
+    public parseEventFromYaml(event: string): EditFormType {
+        const editFormParseValue = parse(event) as EditFormParseType;
+        return editFormParseValue.content.event;
+    }
+
+    public patchFormValues(
+        sectionForm: FormGroup,
+        editFormValue: EditFormType,
+        groupName: SetPollingSourceSection,
+    ): void {
+        switch (groupName) {
+            case SetPollingSourceSection.FETCH: {
+                this.patchFetchStep(sectionForm, editFormValue);
+                break;
+            }
+            case SetPollingSourceSection.READ: {
+                this.patchReadStep(sectionForm, editFormValue);
+                break;
+            }
+            case SetPollingSourceSection.MERGE: {
+                this.patchMergeStep(sectionForm, editFormValue);
+                break;
+            }
+        }
+    }
+
+    public getSetPollingSourceAsYaml(info: DatasetInfo) {
+        return this.appDatasetService
+            .requestDatasetHistory(info, this.historyLength, this.currentPage)
+            .pipe(
+                switchMap(
+                    () => this.appDatasetSubsService.onDatasetHistoryChanges,
+                ),
+                map((h: DatasetHistoryUpdate) => {
+                    this.history = h;
+                    const lastBlock = h.history.filter(
+                        (item: MetadataBlockFragment) =>
+                            item.event.__typename === "SetPollingSource",
+                    )[0];
+                    if (typeof lastBlock === "object") {
+                        return lastBlock.blockHash as string;
+                    }
+                    return h.history[0].blockHash as string;
+                }),
+                switchMap((hash: string) =>
+                    zip(
+                        this.blockService.onMetadataBlockAsYamlChanges,
+                        this.blockService.requestMetadataBlock(info, hash),
+                    ),
+                ),
+            );
+    }
+
+    private patchFetchStep(
+        sectionForm: FormGroup,
+        editFormValue: EditFormType,
+    ): void {
```

Comment:
Very hard to read. Please split on smaller methods

Reply:
Splited on smaller methods

Reply:
Done